doc/btp_spec.txt

```diff
@@ -3760,3 +3772,718 @@ Commands and responses:
+
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Services_Count (1 octet)
+					[array] Included_Service (variable)
+
+		Object Included_Service is defined as:
+					Included_Handle (2 octets)
+					Type (1 octet)
+					Service (7 or 21 octets)
+
+		Valid Type parameter values:
+					0x00 = Primary
+					0x01 = Secondary
+
+		Object Service is defined as:
+					Start_Handle (2 octets)
+					End_Group_Handle (2 octets)
+					UUID_Length (1 octet)
+					UUID (2 or 16 octets)
+
+		Valid UUID_Length parameter values:
+					0x02 = UUID16
+					0x10 = UUID128
+
+		If Status is non-zero Services_Count shall be set to 0.
+
+	Opcode 0x84 - Discover All Characteristics of a Service Response
+
+		Controller Index:	<controller id>
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Characteristics_Count (1 octet)
+					[array] Characteristic (variable)
+
+		Object Characteristic is defined as:
+					Characteristic_Handle (2 octets)
+					Value_Handle (2 octets)
+					Properties (1 octet)
+					UUID_Length (1 octet)
+					UUID (2 or 16 octets)
+
+		Valid UUID_Length parameter values:
+					0x02 = UUID16
+					0x10 = UUID128
+
+		If Status is non-zero Characteristics_Count shall be set to 0.
+
+	Opcode 0x85 - Discover Characteristics by UUID Response
+
+		Controller Index:	<controller id>
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Characteristics_Count (1 octet)
+					[array] Characteristic (variable)
+
+		Object Characteristic is defined as:
+					Characteristic_Handle (2 octets)
+					Value_Handle (2 octets)
+					Properties (1 octet)
+					UUID_Length (1 octet)
+					UUID (2 or 16 octets)
+
+		Valid UUID_Length parameter values:
+					0x02 = UUID16
+					0x10 = UUID128
+
+		If Status is non-zero Characteristics_Count shall be set to 0.
+
+	Opcode 0x86 - Discover All Characteristic Descriptors Response
+
+		Controller Index:	<controller id>
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Descriptors_Count (1 octet)
+					[array] Descriptor (variable)
+
+		Object Descriptor is defined as:
+					Descriptor_Handle (2 octets)
+					UUID_Length (1 octet)
+					UUID (2 or 16 octets)
+
+		Valid UUID_Length parameter values:
+					0x02 = UUID16
+					0x10 = UUID128
+
+		If Status is non-zero Descriptors_Count shall be set to 0.
+
+	Opcode 0x87 - Read Characteristic Value/Descriptor Response
+
+		Controller Index:	<controller id>
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Data_Length (2 octets)
+					Data (variable)
+
+		If Status is non-zero Data_Length shall be set to 0.
+
+	Opcode 0x88 - Read Using Characteristic UUID Response
+
+		Controller Index:	<controller id>
+		Event parameters:	Address_Type (1 octet)
+					Address (6 octets)
+					Status (1 octet)
+					Data_Length (2 octets)
+					Value_Len (1 octet)
```

Comment:
re 1) ATT_READ_BY_TYPE_RSP has common length for all handle-data pairs therefore all must have same length
re 2) I'd expect error response in case it cannot fit into BTP